fem/fespace.cpp

```diff
@@ -1156,33 +1247,74 @@ ::Mult(const Vector &x, Vector &y) const
    Mesh* mesh = fespace->GetMesh();
    const CoarseFineTransformations &rtrans = mesh->GetRefinementTransforms();
 
-   Array<int> dofs, vdofs, old_dofs, old_vdofs;
+   Array<int> dofs, vdofs, old_dofs, old_vdofs, old_Fo;
 
    int vdim = fespace->GetVDim();
    int old_ndofs = width / vdim;
 
-   Vector subY, subX;
+   Vector subY, subX, subYt, subXt;
+
+   //old_VDoFTrans.SetVDim(vdim);
 
    for (int k = 0; k < mesh->GetNE(); k++)
    {
       const Embedding &emb = rtrans.embeddings[k];
       const Geometry::Type geom = mesh->GetElementBaseGeometry(k);
       const DenseMatrix &lP = localP[geom](emb.matrix);
 
-      subY.SetSize(lP.Height());
-
-      fespace->GetElementDofs(k, dofs);
+      DofTransformation *doftrans = fespace->GetElementDofs(k, dofs);
       old_elem_dof->GetRow(emb.parent, old_dofs);
 
-      for (int vd = 0; vd < vdim; vd++)
+      if (!doftrans)
       {
-         dofs.Copy(vdofs);
-         fespace->DofsToVDofs(vd, vdofs);
-         old_dofs.Copy(old_vdofs);
-         fespace->DofsToVDofs(vd, old_vdofs, old_ndofs);
-         x.GetSubVector(old_vdofs, subX);
-         lP.Mult(subX, subY);
-         y.SetSubVector(vdofs, subY);
+         subY.SetSize(lP.Height());
+
+         for (int vd = 0; vd < vdim; vd++)
+         {
+            dofs.Copy(vdofs);
+            fespace->DofsToVDofs(vd, vdofs);
+            old_dofs.Copy(old_vdofs);
+            fespace->DofsToVDofs(vd, old_vdofs, old_ndofs);
+            x.GetSubVector(old_vdofs, subX);
+            lP.Mult(subX, subY);
+            y.SetSubVector(vdofs, subY);
+         }
+      }
+      else
```

Comment:
The dofs that are stored in the `x` and `y` vectors are not necessarily the same dofs that the local `P` matrix was designed for.  It's my understanding that the local `P` matrix assumes it's working with a set of dofs which are meaningful on the reference element. However, the dofs that come straight from `x` and `y` may only be linear combinations of those reference dofs.

The same issue arises now with the signs of the Nedelec dofs.  They may be stored in grid function vectors exactly as the reference element would expect or with some of them multiplied by -1. We currently encode this information into the `vdofs` array so that the `Vector` class can perform the proper transformation when `Vector::Get/SetSubVector` is called. I could have chosen to pass the `DofTransformation` object into `Vector::Get/SetSubVector` to hide it from the caller but I found that hard to justify.  Why require a linear algebra class to handle this FEM issue. Technically we could make the same criticism of flipping the signs inside the `Vector::Get/SetSubVector` methods but in that case I think it's more efficient (and cleaner) to perform that simple transformation inside the `Vector` class.

Regardless of where the transformations are applied the basic issue is the same.  If we applied the local interpolation without first flipping the appropriate signs the resulting interpolation would be incorrect. The same is true of these new `DofTransformation` calls.